contract/models/contract.py

```diff
@@ -81,13 +84,19 @@ class ContractContract(models.Model):
     )
     date_end = fields.Date(compute="_compute_date_end", store=True, readonly=False)
     payment_term_id = fields.Many2one(
-        comodel_name="account.payment.term", string="Payment Terms", index=True
+        comodel_name="account.payment.term",
+        string="Payment Terms",
+        index=True,
+        check_company=True,
+        domain="[('company_id', '=', company_id)]",
```

Comment:
@alexis-via do you believe it is necessary to define this domain? In this case, the objective is to force a company to be defined in each payment_term? In theory, check_company would already check whether the company is false or equal to the company in the record.